openedx/core/djangoapps/content_libraries/api.py

```diff
@@ -1051,3 +1088,271 @@ def revert_changes(library_key):
+    import APIs and can leverage different backends.  It is not aimed towards
+    being a generic API client for Open edX.
+    """
+
+    EXPORTABLE_BLOCK_TYPES = {
+        "drag-and-drop-v2",
+        "problem",
+        "html",
+        "video",
+    }
+
+    def get_block_data(self, block_key):
+        """
+        Get the block's OLX and static files, if any.
+        """
+        raise NotImplementedError()
+
+    def get_export_keys(self, course_key):
+        """
+        Get all exportable block keys of a given course.
+        """
+        raise NotImplementedError()
+
+    def get_block_static_data(self, asset_file):
+        """
+        Get the contents of an asset_file..
+        """
+        raise NotImplementedError()
+
+    def import_block(self, library, modulestore_key):
+        """
+        Import a single modulestore block.
+        """
+
+        # Get or create the block in the library.
+
+        block_data = self.get_block_data(modulestore_key)
+
+        try:
+            library_block = create_library_block(
+                library.library_key,
+                modulestore_key.block_type,
+                modulestore_key.block_id,
+            )
+            blockstore_key = library_block.usage_key
+        except LibraryBlockAlreadyExists:
+            blockstore_key = LibraryUsageLocatorV2(
+                lib_key=library.library_key,
+                block_type=modulestore_key.block_type,
+                usage_id=modulestore_key.block_id,
+            )
+            get_library_block(blockstore_key)
+
+        # Handle static files.
+
+        for filename, static_file in block_data.get('static_files', {}).items():
+            files = [
+                f.path for f in
+                get_library_block_static_asset_files(blockstore_key)
+            ]
+            if filename in files:
+                # Files already added, move on.
+                continue
+            file_content = self.get_block_static_data(static_file)
+            add_library_block_static_asset_file(
+                blockstore_key, filename, file_content)
+
+        # Import OLX and publish.
+
+        set_library_block_olx(blockstore_key, block_data['olx'])
+        publish_changes(blockstore_key.lib_key)
+
+
+class EdxModulestoreImportClient(BaseEdxImportClient):
+    """
+    An import client based on the local instance of modulestore.
+    """
+
+    def __init__(self, modulestore_instance=None):
+        """
+        Initialize the client with a modulestore instance.
+        """
+        self.modulestore = modulestore_instance or modulestore()
+
+    def get_block_data(self, block_key):
+        """
+        Get block OLX by serializing it from modulestore directly.
+        """
+        block = self.modulestore.get_item(block_key)
+        data = XBlockSerializer(block)
+        return {'olx': data.olx_str,
+                'static_file': {s.name: s for s in data.static_files}}
```

Comment:
@jvdm, this has the same problem I fixed [here](https://github.com/open-craft/edx-platform/pull/378/commits/b6a771ee24ee4f653c0cc637f4255c0bf49836bb).  It should be "static_files".

Reply:
Created a [PR to this PR](https://github.com/open-craft/edx-platform/pull/379) to address the issue.